local/sdk-sriov/pkg/k8s/manager_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+import (
+	"context"
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+	podresources "k8s.io/kubernetes/pkg/kubelet/apis/podresources/v1alpha1"
+
+	testingtools "github.com/networkservicemesh/cmd-forwarder-sriov/test/tools"
+)
+
+var devicePluginPath = path.Join(os.TempDir(), "device-plugins")
+var devicePluginSocket = path.Join(devicePluginPath, kubeletSocket)
+
+func TestDevicePluginManager_MonitorKubeletRestart(t *testing.T) {
+	dpm := NewManager(devicePluginPath, "")
+
+	_ = os.RemoveAll(devicePluginPath)
+	err := os.Mkdir(devicePluginPath, os.ModeDir|os.ModePerm)
+	assert.Nil(t, err)
+
+	monitorCh, err := dpm.MonitorKubeletRestart(context.TODO())
+	assert.Nil(t, err)
+
+	_, err = os.Create(devicePluginSocket)
+	assert.Nil(t, err)
+
+	_, ok := testingtools.ReadBoolChan(t, monitorCh, 10*time.Second)
```

Comment:
Please take a look at tesitfy.Eventually

Reply:
I think 10 seconds to wait is long. Let's use a 1 second, as this is a unit test that cannot have external factors.

Reply:
I agree for the 1 second, but `assert.Eventually(...)` doesn't looks great for such case, because I want to get the results from the channel and so make some further assertions.
```
var ok bool
assert.Eventually(t, func() {
    _, ok = <-monitorCh
    return true
}, time.Second, time.Second)
```
it is not something like
```
_, ok := testingtools.ReadBoolChan(t, monitorCh, time.Second)
```

Reply:
Why do we can't just use select in the test body? I think it will be more clear than `testingtools.ReadBoolChan`

```go
	select {
	case val, ok = <-ch:
		// do assert here
	case <-time.After(timeout):
		// fail test
	}
```

Reply:
Same thing - I don't like it :)
```
var ok bool
select {
case <-time.After(time.Second):
    assert.FailNow(...)
case _, ok = <-ch:
}
```
It super clear I suppose. What else could it be if not "read bool chan with timeout"?
```
_, ok := testingtools.ReadBoolChan(t, monitorCh, time.Second)
```

Reply:
For the golang users `select` block is a not a problem :)

Reply:
I think this point is not supercritical and can be ignored. My point is using natural syntax for improving readability for other developers.